func.go

```diff
@@ -134,6 +135,9 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 					stack++
 				}
 			case reflect.Float32, reflect.Float64:
+				if is32bit() {
+					panic("purego: floats only supported on 64bit platforms")
```

Comment:
Is this panic related to returning structs?

Reply:
It's not. I had just realized it would just silently fail on 32bit performs. I can separate that if you want

Reply:
Yeah, please separate this. The smaller the change is, the easier we can review.

Reply:
So, will you be sending the PR for this part now, or later?